File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    // Memoization
    public int climbStairs(int n) {
        int[] memo = new int[n + 1];
        return dp(n , memo);
    }
    public int dp(int n , int[] memo){
        if(n <= 0) return 0;
        if(n == 1) return 1;
        if(n == 2) return 2;
        if(memo[n] != 0) return memo[n];
        int x = dp(n-1 , memo) + dp(n-2 , memo);
        memo[n] = x;
        return x;
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1: 
            return 1
        last = 1
        beforelast = 2
        for i in range(n, 2, -1):
            temp = beforelast + last
            last = beforelast
            beforelast = temp
        return beforelast
            
        